ArduCopter/control_althold.cpp

```diff
@@ -57,19 +57,17 @@ void Copter::althold_run()
 
 #if FRAME_CONFIG == HELI_FRAME
     // helicopters are held on the ground until rotor speed runup has finished
-    bool takeoff_triggered = (ap.land_complete && (channel_throttle->get_control_in() > get_takeoff_trigger_throttle()) && motors.rotor_runup_complete());
+    bool takeoff_triggered = (ap.land_complete && (target_climb_rate > 0.0f) && motors.rotor_runup_complete());
 #else
-    bool takeoff_triggered = (ap.land_complete && (channel_throttle->get_control_in() > get_takeoff_trigger_throttle()) && motors.spool_up_complete());
+    bool takeoff_triggered = ap.land_complete && (target_climb_rate > 0.0f);
 #endif
 
     // Alt Hold State Machine Determination
     if (!motors.armed() || !motors.get_interlock()) {
         althold_state = AltHold_MotorStopped;
-    } else if (!ap.auto_armed){
-        althold_state = AltHold_NotAutoArmed;
-    } else if (takeoff_state.running || takeoff_triggered){
+    } else if (takeoff_state.running || takeoff_triggered) {
         althold_state = AltHold_Takeoff;
-    } else if (ap.land_complete){
+    } else if (!ap.auto_armed || ap.land_complete) {
```

Comment:
You're probably right.  It does no harm in most cases but perhaps not much good.  We always use this auto_armed as a double check that the pilot has raised the throttle at least once.  I've discussed with @jschall and he's going to do a PR to remove auto-armed completely.